.github/workflows/ci.yml

```diff
@@ -0,0 +1,84 @@
+# Inspired by worfklows in https://github.com/dalek-cryptography/curve25519-dalek/
+name: CI
+
+on:
+  push:
+  pull_request:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  doc:
+    name: Check docs
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/checkout@v3
+      - run: rustup default stable
+      - run: cargo doc
+
+  build:
+    name: Build the code
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: rustup default stable
+      - run: cargo build
+
+  test:
+    name: Run tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - run: rustup default stable
+      - run: cargo test
+
+  prove:
+    name: Run crux tests and proofs
+    runs-on: ubuntu-24.04
+    steps:
+      - uses: actions/checkout@v3
+      - name: Install z3
+        # NOTE: the following does not run on Ubuntu 22.04
+        run: |
+          wget https://github.com/Z3Prover/z3/releases/download/z3-4.15.0/z3-4.15.0-x64-glibc-2.39.zip
+          unzip z3-4.15.0-x64-glibc-2.39.zip
+          echo "${PWD}/z3-4.15.0-x64-glibc-2.39/bin" >> $GITHUB_PATH
+      - name: Test z3
+        run: z3 --version
+
+      - name: Get the URL for the latest mir-json build
+        id: mir-json-build
+        run: |
+          echo "DOWNLOAD_URL=$(curl -L -H "Accept: application/vnd.github+json" -H "Authorization: Bearer ${{ secrets.CRUX_ACCESS_TOKEN }}" -H "X-GitHub-Api-Version: 2022-11-28" https://api.github.com/repos/galoisinc/mir-json/actions/artifacts | jq -r '.artifacts' | jq -c '[ .[] | select ( .name | contains("ubuntu") ) ][0].archive_download_url')" >> $GITHUB_OUTPUT
+
+      - name: Download the latest mir-json build
+        run: |
+          echo "Downloading from ${{ steps.mir-json-build.outputs.DOWNLOAD_URL }}"
+          curl -L -H "Accept: application/vnd.github+json" -H "Authorization: Bearer ${{ secrets.CRUX_ACCESS_TOKEN }}" -H "X-GitHub-Api-Version: 2022-11-28" --output mir-json.zip ${{ steps.mir-json-build.outputs.DOWNLOAD_URL }}
```

Comment:
This seems quite verbose.  Would it make sense to either move this into a shell script, or use the [download-artifact action](https://github.com/actions/download-artifact?tab=readme-ov-file#download-artifacts-from-other-workflow-runs-or-repositories)?

Reply:
I reused the approach that [Verus takes](https://github.com/verus-lang/verus/discussions/1541#discussioncomment-12646417) - to make more transparent what is happening, but I agree that it is a handful.

Once we have releases of `crux-mir` and `mir-json` that work together (not the case currently) then I can just download the release rather than CI artifacts, which will be much simpler.

Reply:
OK, that seems fine